pkgs/http/lib/src/utils.dart

```diff
@@ -12,14 +12,11 @@ import 'byte_stream.dart';
 ///
 ///     mapToQuery({"foo": "bar", "baz": "bang"});
 ///     //=> "foo=bar&baz=bang"
-String mapToQuery(Map<String, String> map, {Encoding? encoding}) {
-  var pairs = <List<String>>[];
-  map.forEach((key, value) => pairs.add([
-        Uri.encodeQueryComponent(key, encoding: encoding ?? utf8),
-        Uri.encodeQueryComponent(value, encoding: encoding ?? utf8)
-      ]));
-  return pairs.map((pair) => '${pair[0]}=${pair[1]}').join('&');
-}
+String mapToQuery(Map<String, String> map, {Encoding encoding = utf8}) =>
```

Comment:
Since this is only used in one place and the encoding is provided, you con probably change this to:

```dart
... {required Encoding encoding})
```